Extension/src/LanguageServer/configurations.ts

```diff
@@ -909,20 +909,18 @@ export class CppProperties {
             if (!configuration.browse.path) {
                 if (settings.defaultBrowsePath) {
                     configuration.browse.path = settings.defaultBrowsePath;
-                } else if (configuration.includePath) {
-                    // If the user doesn't set browse.path, copy the includePath over. Make sure ${workspaceFolder} is in there though...
-                    configuration.browse.path = configuration.includePath.slice(0);
-                    if (configuration.includePath.findIndex((value: string, index: number) =>
-                        !!value.match(/^\$\{(workspaceRoot|workspaceFolder)\}(\\\*{0,2}|\/\*{0,2})?$/g)) === -1
-                    ) {
-                        configuration.browse.path.push("${workspaceFolder}");
-                    }
+                } else {
+                    // If the user doesn't set browse.path, make sure ${workspaceFolder} is in there though...
+                    configuration.browse.path = ["${workspaceFolder}"];
                 }
             } else {
                 configuration.browse.path = this.updateConfigurationPathsArray(configuration.browse.path, settings.defaultBrowsePath, env);
             }
 
             configuration.browse.limitSymbolsToIncludedHeaders = this.updateConfigurationStringOrBoolean(configuration.browse.limitSymbolsToIncludedHeaders, settings.defaultLimitSymbolsToIncludedHeaders, env);
+            if (configuration.browse.limitSymbolsToIncludedHeaders !== false) {
+                configuration.browse.limitSymbolsToIncludedHeaders = true;
```

Comment:
This should already be defaulting to `true`. The problem you may be seeing is that our include iteration code only checks the filenames and not the complete paths that is used when combined with the includePath, i.e. `#include <string.h>` would include symbols for all "string.h" files found and not the one that is chosen based on the includePath.